zebra/zebra_nhg.h

```diff
@@ -203,10 +234,14 @@ extern int zebra_nhg_kernel_find(uint32_t id, struct nexthop *nh,
 /* Del via kernel */
 extern int zebra_nhg_kernel_del(uint32_t id, vrf_id_t vrf_id);
 
-/* Find via route creation */
+/* Find an nhe based on a nexthop_group */
 extern struct nhg_hash_entry *
 zebra_nhg_rib_find(uint32_t id, struct nexthop_group *nhg, afi_t rt_afi);
 
+/* Find an nhe based on a route's nhe, used during route creation */
+struct nhg_hash_entry *
+zebra_nhg_rib_find_nhe(struct nhg_hash_entry *rt_nhe, afi_t rt_afi);
```

Comment:
It seems like by adding this second find path from the rib-side of things (either zebra_rib or zapi_message), we are duplicating the code path a bit to handle the backup nexthop.

Would it not be better to just extend `zebra_nhg_rib_find()` to handle backup nexthops as an additional argument?

Reply:
I wanted to be able to use the nhe, which can hold an id, a list of nexthops, and a list of backups - that seemed more compact than having to use one of the other apis that basically take all the pieces, use them to assemble an nhe, and then probe using that nhe. I figure that as this evolves and we see the patterns that really make sense, some of the paths will be trimmed out and converged.

Reply:
Fair point.

Would it be better to just remove the other `zebra_nhg_rib_find()` API then altogether and convert them to use this one?

Reply:
I didn't think so - it seemed ok to have different ways to look things up (as long as we're sharing the actual logic).